packages/fiori/cypress/specs/UploadCollection.cy.tsx

```diff
@@ -377,24 +460,166 @@ describe("Keyboard handling", () => {
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-li-detailbtn")
+			.realClick();
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-input-inner", { includeShadowDom: true })
+			.should("have.focus")
+			.type("last.reports-edited");
+
+		cy.realPress("Enter");
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-uci-file-name")
+			.should("have.text", "last.reports-edited");
 	});
 
 	it("Tests that extension is able to be added, if there isn't such", () => {
+		const newFileName = "newFileName.newExtension";
+		const newFileName2 = "newFileName2";
+
+		cy.mount(
+			<UploadCollection id="uploadCollection">
+				<UploadCollectionItem id="item" fileName="File name" type="Detail" />
+			</UploadCollection>);
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-li-detailbtn")
+			.realClick();
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-input-inner", { includeShadowDom: true })
+			.should("have.focus")
+			.type(newFileName);
+
+		cy.realPress("Enter");
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-uci-file-name")
+			.should("have.text", newFileName);
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-li-detailbtn")
+			.realClick();
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-input-inner", { includeShadowDom: true })
+			.should("have.focus")
+			.type(newFileName2);
+
+		cy.realPress("Enter");
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-uci-file-name")
+			.should("have.text", newFileName2 + ".newExtension");
 	});
 
 	it("Tests that hidden file name is NOT considered as extension", () => {
+		cy.mount(
+			<UploadCollection id="uploadCollection">
+				<UploadCollectionItem id="item" file-name=".gitignore" type="Detail" />
+			</UploadCollection>);
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-li-detailbtn")
+			.realClick();
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-uci-file-name")
+			.should("not.have.text");
 	});
 
 	it("Tests cancelling of name change via keyboard", () => {
+		cy.mount(
+			<UploadCollection id="uploadCollection">
+				<UploadCollectionItem id="item" file-name=".gitignore" type="Detail" />
+			</UploadCollection>);
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-li-detailbtn")
+			.realClick();
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-input-inner", { includeShadowDom: true })
+			.should("have.focus")
+			.type("new name");
+
+		cy.realPress("Tab");
+		cy.realPress("Tab");
+		cy.realPress("Enter");
+
+		cy.get("#item")
+			.shadow()
+			.find(".ui5-uci-file-name")
+			.should("have.text", ".gitignore");
 	});
 });
 
 describe("Drag and Drop", () => {
 	it("Tests that drag and drop overlay is NOT shown when NOT dragging files", () => {
+		cy.mount(
+			<>
+				<UploadCollection id="uploadCollection" />
+				<p id="draggableElement" draggable="true">This element is draggable</p>
+			</>
+		);
+
+		cy.get("#draggableElement").scrollIntoView();
+
+		const dataTransfer = new DataTransfer();
+
+		cy.get("#draggableElement").trigger("dragstart", { dataTransfer, bubbles: true });
```

Comment:
we can enhance the test to verify that the overlay is shown when drag has started